### DISCUSSION TITLE
Any pointers in adding accessory logs?

### DISCUSSION THREAD

**Discussion by frankonski** [HomeSpan/HomeSpan#658] [Q&A]
I took it upon myself to try to get accessory logs to work. [HMCharacteristicTypeLogs](https://developer.apple.com/documentation/homekit/hmcharacteristictypelogs) seems to be a characteristic available part of **AccessoryInformation**.

I'm able to add a new OPT characteristic, its ID being **1F** if I got this right, and the format to send back to a Home device is, i think, a set of TLV8. 

From my research I'd only have to base64-encode (<one byte for the string type><one byte for the length of text><text to log>).

I'm trying to add logging to a SecuritySystem accessory, so that from the Home app, I can go to the accessory itself and see logs I want to push, like zones being open & all.

Looking for guidance and suggestions. Should I push what I have done in a branch (in my own repo) for review / help?

**Comment by HomeSpan** [comment]
The Log Characteristic requires TLV8 data type.   This is the only data type I have *not* implemented in HomeSpan.  It's doable, but would require a number of extensions to SpanCharacteristic.

**Comment by frankonski** [comment]
I realize my comment above is not displaying what should be encoded:
* one byte for the string type: I am not sure the value to use for this byte though
* one byte for the length of text
* text to log

I'd initially try to get the HOme app to display that Log option, which is not yet happening... Even if this is an empty set of TLV8 values.

**Reply by frankonski** [reply]
well, it looks like Apple decided to make my life easier. With iOS 17, the Home app now has the built-in ability to keep 1 month of "Activity History" for SecuritySystem accessories.
Ran some initial tests, alarm triggered gets logged, open doors get logged. Motion sensors do not get logged.
My goal is to be able to see "what triggered the alarm". With this new iOS feature, I will be able to know that a _Door Open_ even triggered the alarm when looking at that new Activity History. But if a motion sensor triggers the alarm, I won't know it caused it,
